### PR TITLE
Replace Vector2D with GeoCoord.

### LINKED ISSUE #16: h3ToGeo instantiates Vector2D incorrectly
```
    public Vector2D h3ToGeo(long h3) {
        double[] coords = new double[2];
        h3Api.h3ToGeo(h3, coords);
        Vector2D out = new Vector2D(
                constrainLat(toDegrees(coords[0])),
                constrainLng(toDegrees(coords[1]))
        );
        return out;
    }
```
Is incorrect. Vector2d is defined as - `Vector2d(double x, double y)`

**Lat should be y and Lng should be x**

Tested using the following groovy script 
```
import com.uber.h3core.H3Core

H3Core h3 = H3Core.newInstance()
int res = 8
double bigBenLat = 51.5007292
double bigBenLon = -0.1246254

long index = h3.geoToH3(bigBenLat, bigBenLon, res)

double reasonableTolerance = 0.5
assert h3.h3ToGeo(index).y < (bigBenLat + reasonableTolerance) && h3.h3ToGeo(index).y > (bigBenLat + reasonableTolerance)  :
        "Oh no! The index is a long way from Big Ben. The Big Ben lat is $bigBenLat and h3 index centre is ${h3.h3ToGeo(index).y}"
assert h3.h3ToGeo(index).x < (bigBenLon + reasonableTolerance) && h3.h3ToGeo(index).x > (bigBenLon + reasonableTolerance)  :
        "Oh no! The index is a long way from Big Ben. The Big Ben lat is $bigBenLon and h3 index centre is ${h3.h3ToGeo(index).x}"
```

which gets the following stack trace 

```
Caught: java.lang.AssertionError: Oh no! The index is a long way from Big Ben. The Big Ben lat is 51.5007292 and h3 index centre is -0.1260662704929132. Expression: ((h3.h3ToGeo(index).y < (bigBenLat + reasonableTolerance)) && (h3.h3ToGeo(index).y > (bigBenLat + reasonableTolerance))). Values: bigBenLat = 51.5007292, reasonableTolerance = 0.5, bigBenLat = 51.5007292, reasonableTolerance = 0.5
java.lang.AssertionError: Oh no! The index is a long way from Big Ben. The Big Ben lat is 51.5007292 and h3 index centre is -0.1260662704929132. Expression: ((h3.h3ToGeo(index).y < (bigBenLat + reasonableTolerance)) && (h3.h3ToGeo(index).y > (bigBenLat + reasonableTolerance))). Values: bigBenLat = 51.5007292, reasonableTolerance = 0.5, bigBenLat = 51.5007292, reasonableTolerance = 0.5
```

### PR DESCRIPTION
**Pull request by isaacbrodsky**
This better matches the C library, and eliminates options about coordinate ordering.

Fixes #16.

Edit: note that this is a breaking change.

cc @dfellis @nrabinowitz @joegilley

### CODE REVIEW

**Comment by isaacbrodsky** [comment]
Restarted the CI jobs because the Java 8 ones hit the maximum time limit. Not sure if that's a hang bug or just they ran slow.

File: src/main/java/com/uber/h3core/H3Core.java [hunk 1]
```diff
@@ -487,22 +463,14 @@ public Vector2D h3ToGeo(String h3Address) {
      *
      * @return Next offset to begin filling from
      */
-    private static int packGeofenceVertices(double[] arr, List<Vector2D> original, int offset, boolean geoJsonOrder) {
+    private static int packGeofenceVertices(double[] arr, List<GeoCoord> original, int offset) {
```
[the remark this answers is not in the document]
Comment (isaacbrodsky) [reply src/main/java/com/uber/h3core/H3Core.java]: It seems we didn't care about it before. If we think we're going to care about it later, we should not change this API.

**Comment by isaacbrodsky** [comment]
@nrabinowitz Yes, I think this would need to be versioned as 3.0.3. It is unfortunate.

There are some other options such as appending `-api2` or something to the version identifier, Less drastic than that is to somehow tag/annotate the release as being a breaking version. This doesn't help if you go from 3.0.2 to 3.0.4, but could at least make the changelog clearer.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ file [H3Core.java](./src/main/java/com/uber/h3core/H3Core.java), and support
 for the Linux x64 and Darwin x64 platforms.
 
 ## [Unreleased]
+### Changed
+- Replaced Vector2D with GeoCoord.
```
Comment (joegilley) [inline CHANGELOG.md]: i might comment that this change is breaking, despite being a patch update.

Comment (isaacbrodsky) [reply CHANGELOG.md]: I changed this to note both changes are breaking.

**Pull request merged**